test/src/test/java/jenkins/security/Security3135Test.java

```diff
@@ -0,0 +1,49 @@
+package jenkins.security;
+
+import static org.junit.Assert.assertNull;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.InvisibleAction;
+import hudson.model.UnprotectedRootAction;
+import org.htmlunit.html.DomElement;
+import org.htmlunit.html.HtmlElement;
+import org.htmlunit.html.HtmlPage;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.TestExtension;
+
+public class Security3135Test {
+    public static final String ACTION_URL = "security3135";
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    @Issue("SECURITY-3135")
+    public void contextMenuShouldNotBypassCSRFProtection() throws Exception {
+        JenkinsRule.WebClient wc = j.createWebClient();
+        FreeStyleProject project = j.createFreeStyleProject("FreestyleProject");
+
+        HtmlPage page = wc.goTo(ACTION_URL);
+        DomElement anchor = page.getElementById("context-menu");
+        HtmlElement nextSibling = (HtmlElement) anchor.getFirstChild().getNextSibling();
```

Comment:
This test looks unconvincing to me, in particular it looks like it would continue to pass with UI rework resulting in the `click`ed element being one without associated behaviors. Some options:

* Assert the HTTP request/response as a result of the interaction.
* Have a corresponding "good case" case with the same.
* Assert something about this element, so the test starts failing during UI rework.

Additionally, commenting out https://github.com/jenkinsci/jenkins/blob/8041e14c42084930c7b8ade2b7ffb2b2b85576b4/core/src/main/java/jenkins/model/ParameterizedJobMixIn.java#L211 should fail the test (AFAICT), but does not. Interactive testing indicates that aggregated URL contains an unparseable delay string due to concatenation, so results in HTTP 500 (`java.lang.IllegalArgumentException: Invalid time duration value: 0sec/contextMenu?delay=0`).